react/components/Modal/index.test.js

```diff
@@ -0,0 +1,271 @@
+
+      render(
+        <Modal isOpen onClose={onClose}>
+          Foo
+        </Modal>
+      )
+      fireEvent.click(document.querySelector('.vtex-modal__overlay'))
+      expect(onClose).toHaveBeenCalled()
+    })
+
+    it('should not be called when closeOnOverlayClick prop is false', () => {
+      const onClose = jest.fn()
+
+      render(
+        <Modal isOpen onClose={onClose} closeOnOverlayClick={false}>
+          Foo
+        </Modal>
+      )
+      fireEvent.click(document.querySelector('.vtex-modal__overlay'))
+      expect(onClose).not.toHaveBeenCalled()
+    })
+  })
+
+  describe('title', () => {
+    it('should render title prop', () => {
+      const containerModal = document.createElement('div')
+      const onClose = jest.fn()
+      const title = 'modal title'
+
+      const { getByText, asFragment } = render(
+        <Modal
+          isOpen
+          onClose={onClose}
+          title={title}
+          container={containerModal}
+        >
+          Foo
+        </Modal>,
+        { container: containerModal }
+      )
+      expect(getByText(title)).not.toBeNull()
+      expect(asFragment()).toMatchSnapshot()
+    })
+  })
+
+  describe('isOpen', () => {
+    it('Should show modal when isOpen is true and hide when is false', () => {
+      const onClose = jest.fn()
+      const containerModal = document.createElement('div')
+
+      const { rerender, container } = render(
+        <Modal isOpen onClose={onClose} container={containerModal}>
+          Foo
+        </Modal>,
+        { container: containerModal }
+      )
+      expect(container).toMatchSnapshot()
+
+      rerender(
+        <Modal isOpen={false} onClose={onClose} container={containerModal}>
+          Foo
+        </Modal>,
+        { container: containerModal }
+      )
+      expect(container).toMatchSnapshot()
+    })
+  })
+
+  describe('bottomBar', () => {
+    it('should render bottomBar', () => {
+      const onClose = jest.fn()
+      const containerModal = document.createElement('div')
+
+      const { getByText, container } = render(
+        <Modal
+          isOpen
+          onClose={onClose}
+          bottomBar={<div>Bar</div>}
+          container={containerModal}
+        >
+          Foo
+        </Modal>,
+        { container: containerModal }
+      )
+
+      expect(getByText('Bar')).not.toBeNull()
+      expect(container).toMatchSnapshot()
+    })
+  })
+
+  describe('container', () => {
+    it('should render modal inside it', () => {
+      const onClose = jest.fn()
+      const containerModal = document.createElement('div')
+      const { container } = render(
+        <Modal isOpen onClose={onClose} container={containerModal}>
+          Foo
+        </Modal>,
+        { container: document.body.appendChild(containerModal) }
+      )
+      expect(container.querySelector('.vtex-modal__modal')).not.toBeNull()
+      expect(container).toMatchSnapshot()
+    })
+  })
+  // help it's not working
```

Comment:
Remove code and open issue.

Reply:
I'm gonna wait to see if someone suggests a fix for this test here in the PR, if not I'll remove it

Reply:
Fixed!